### PR TITLE
Testing/dashboard

### PR DESCRIPTION
**Pull request by asodemann18**
# PR Description
Please include a summary of the change:
- Added testing to dashboard component

Commit Emojis:
:tada: Initial commit
:construction: HTML modifications/updates.
:art: CSS modifications/updates.
:crystal_ball: JS modifications/updates.
:recycle: refactor code.
:camera_with_flash: adding screenshots.
:bento: Adding new assets or images.
:beetle: If it is a bug or issue fix, describe the issue that was fixed.
:mute: remove logs and empty space/linter.
:bar_chart: new tests
:white_check_mark: add or update tests

If it is a new feature, describe the feature and the iteration of the project spec it addresses.

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (non-breaking change which adds testing)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code follows the Turing style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/apiCalls.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const getAllCocktails = async (): Promise<AllCocktailsDetails> => {
   return data;
 };
 
-export const getRandomCocktail = async ():Promise< { [key: string]: string }> => {
+export const getRandomCocktail = async ():Promise<RandomCocktail> => {
```
Comment (Melizzo) [inline src/apiCalls.tsx:17]: I saw this in apiCalls and was wondering what the deleted file was doing.  But it appears that it needs to be interface for the expected object -very cool but still curious about what the deleted line was doing.

File: src/Dashboard/Dashboard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import Dashboard from "./Dashboard";
+import "@testing-library/jest-dom";
+import { render, waitFor } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+import { getRandomCocktail } from "../apiCalls";
+import { mocked } from "ts-jest/utils";
+jest.mock("../apiCalls");
+
+describe("Dashboard", () => {
+  mocked(getRandomCocktail).mockImplementation(() =>
```
Comment (Melizzo) [inline src/Dashboard/Dashboard.test.tsx:11]: is mockImplementation a keywork for TS Jest?

**Review by Melizzo — APPROVED** [review]

File: src/apiCalls.tsx (hunk 1 above)
Comment (asodemann18) [reply src/apiCalls.tsx:17]: It technically bypasses the strictness of TS (so isn't recommended) and was a fix that Becca had found to get the removeNulls functionality working but that's been put on the backburner and Hannah helped us get a solution for it that was following TS's rules.

File: src/Dashboard/Dashboard.test.tsx (hunk 2 above)
Comment (asodemann18) [reply src/Dashboard/Dashboard.test.tsx:11]: That's how I understood it

File: src/apiCalls.tsx (hunk 1 above)
Comment (Melizzo) [reply src/apiCalls.tsx:17]: I get it - saying that there will be an object returned that has a key/value that really can be anything!  Thanks Alex.

File: src/Dashboard/Dashboard.test.tsx (hunk 2 above)
Comment (st0nebreaker) [reply src/Dashboard/Dashboard.test.tsx:11]: Very cool!

**Comment by st0nebreaker** [comment]
Good job fixing that TS error in the test suite, Alex

**Pull request merged**